osu.Game/Rulesets/Objects/Legacy/ConvertHitObjectParser.cs

```diff
@@ -349,16 +351,19 @@ private List<HitSampleInfo> convertSoundType(LegacyHitSoundType type, SampleBank
                 };
             }
 
-            var soundTypes = new List<HitSampleInfo>
+            var soundTypes = new List<HitSampleInfo>();
+
+            if (type.HasFlag(LegacyHitSoundType.Normal) || AllowsLayeredHitSounds)
             {
-                new LegacyHitSampleInfo
+                soundTypes.Add(new LegacyHitSampleInfo
                 {
                     Bank = bankInfo.Normal,
                     Name = HitSampleInfo.HIT_NORMAL,
                     Volume = bankInfo.Volume,
-                    CustomSampleBank = bankInfo.CustomSampleBank
-                }
-            };
+                    CustomSampleBank = bankInfo.CustomSampleBank,
+                    IsLayered = !type.HasFlag(LegacyHitSoundType.Normal)
```

Comment:
I feel like this is backwards. Shouldn't this just use `AllowsLayeredHitSounds`? If both the flag is set and `AllowsLayeredHitSounds` is `true`, won't this wrongly consider the sample non-layered?

Reply:
The thing is that the beatmap could actually explicitly indicate that the sample to be played (among others) should be hitnormal. In this case, whether `AllowLayeredHitSounds` is `true` or `false` should not affect the fact that the sample is not layered. That means it should always be played even in osu!mania. That's why this problem doesn't appear in other beatmap as they use hitnormal as opposed to this map that uses hitfinish and the others but not hitnormal.
I'm actually concerned about the naming of this flag, maybe `IsForced` would make more sense?

Reply:
With application of the above comments, the outer condition can once again be removed and this can remain here.